framework/docs/source/tutorial-series-use-a-federated-learning-strategy-pytorch.rst

```diff
@@ -0,0 +1,421 @@
+the model to clients. We’re also fortunate in the sense that our entire evaluation
+dataset is available at all times.
+
+**Federated Evaluation** (or *client-side evaluation*) is more complex, but also more
+powerful: it doesn’t require a centralized dataset and allows us to evaluate models over
+a larger set of data, which often yields more realistic evaluation results. In fact,
+many scenarios require us to use **Federated Evaluation** if we want to get
+representative evaluation results at all. But this power comes at a cost: once we start
+to evaluate on the client side, we should be aware that our evaluation dataset can
+change over consecutive rounds of learning if those clients are not always available.
+Moreover, the dataset held by each client can also change over consecutive rounds. This
+can lead to evaluation results that are not stable, so even if we would not change the
+model, we’d see our evaluation results fluctuate over consecutive rounds.
+
+We’ve seen how federated evaluation works on the client side (i.e., by implementing the
+``evaluate`` method in ``FlowerClient``). Now let’s see how we can evaluate aggregated
+model parameters on the server-side. First we define a new function ``evaluate`` in
+``task.py``:
+
+.. code-block:: python
+
+    # The `evaluate` function will be called by Flower after every round
+    def evaluate(
+        server_round: int,
+        parameters,
+        config,
+    ):
+        device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+        net = Net().to(device)
+        _, testloader = load_data(0, 10)
+        set_weights(net, parameters)  # Update model with the latest parameters
+        loss, accuracy = test(net, testloader, device)
+        print(f"Server-side evaluation loss {loss} / accuracy {accuracy}")
+        return loss, {"accuracy": accuracy}
+
+Next, in ``server_app.py``, we pass the ``evaluate`` function to the ``evaluate_fn``
+parameter of the ``FedAvg`` strategy:
+
+.. code-block:: python
+
+    def server_fn(context: Context) -> ServerAppComponents:
+        # Read from config
+        num_rounds = context.run_config["num-server-rounds"]
+        fraction_fit = context.run_config["fraction-fit"]
+
+        # Initialize model parameters
+        ndarrays = get_weights(Net())
+        parameters = ndarrays_to_parameters(ndarrays)
+
+        strategy = FedAvg(
+            fraction_fit=fraction_fit,
+            fraction_evaluate=1.0,
+            min_available_clients=2,
+            initial_parameters=parameters,
+            evaluate_fn=evaluate,
+        )
+        config = ServerConfig(num_rounds=num_rounds)
+
+        return ServerAppComponents(strategy=strategy, config=config)
+
+
+    # Create ServerApp
+    app = ServerApp(server_fn=server_fn)
+
+Finally, we run the simulation.
+
+.. code-block:: shell
+
+    $ flwr run .
+
+Sending/receiving arbitrary values to/from clients
```

Comment:
I'd rename this to something like: Sending configs to clients from the strategies. Using "arbitrary", although possible if objs are serialized, is maybe a bit too generous for what's covered in an introductory tutorial.